scipy/stats/tests/test_sampling.py

```diff
@@ -835,3 +839,91 @@ def pdf(self, x):
         msg = r"Exact CDF required but not found."
         with pytest.raises(ValueError, match=msg):
             rng.u_error()
+
+
+class Gamma:
+    def __init__(self, p):
+        self.p = p
+
+    def pdf(self, x):
+        if x < 0:
+            return 0.0
+        return x**(self.p - 1) * math.exp(-x)
+
+    def cdf(self, x):
+        return stats.gamma.cdf(x, self.p)
+
+    @staticmethod
+    def support():
+        return 0, np.inf
+
+
+class TestNaiveRatioUniforms:
```

Comment:
I think the tests are already quite strong. If you want, you can also add the distributions from UNU.RAN's test file `t_nrou.c`. I looked at it and it tests with the `dist2` distribution present in `TransformedDensityRejection`'s test suite (on line 336 in `test_sampling.py`).